pkg/reduce/readloop/readloop.go

```diff
@@ -362,6 +366,9 @@ func (rl *ReadLoop) closePartitions(partitions []partition.ID) {
 	for _, p := range partitions {
 		q := rl.pbqManager.GetPBQ(p)
 		rl.log.Infow("Close of book", zap.String("partitionID", p.String()))
+		rl.op.insertTask(rl.udfInvocationTracking[p])
+		partitionsInFlight.With(map[string]string{metrics.LabelVertex: rl.op.vertexName, metrics.LabelPipeline: rl.op.pipelineName}).Inc()
 		q.CloseOfBook()
```

Comment:
why are we incrementing partitionsInFlight while closing the partition? merge conflict?

Reply:
i moved it here, but i guess i can leave it there itself.

Reply:
since this is streaming, it makes sense to increment this as soon as the go routine is scheduled.

Reply:
yeah